File: 2021/day-01/day01-1.py
file = open('input.txt', 'r')
lines = file.readlines()
 
count = 0
previous = -1
for line in lines:
    current = int(line)
    if previous != -1 and current > previous:
        count += 1
    
    previous = current

print("{}".format(count))

File: 2021/day-01/day01-1.cpp
#include <iostream>
#include <fstream>
#include <climits>

int main(int argc, char** argv)
{
    int count = 0;
    int previous = INT_MIN;
    int current = 0;
    std::ifstream file("input.txt");
    file >> previous;
    while (file >> current)
    {
        if (current > previous)
        {
            ++count;
        }

        previous = current;
    }

    std::cout << count << std::endl;

    return 0;
}
